src/amo/utils/index.js

```diff
@@ -97,3 +98,24 @@ export const getPromotedBadgesLinkUrl = ({
     utm_content,
   })}`;
 };
+
+export const stripLangFromAmoUrl = ({
+  _checkInternalURL = checkInternalURL,
+  urlString,
+}: {
+  _checkInternalURL?: typeof checkInternalURL,
+  urlString: string,
+}) => {
+  if (_checkInternalURL({ urlString }).isInternal) {
+    const urlParts = urlString.split('?');
+    const urlSlashSeparated = urlParts[0].split('/');
+
+    for (const part of urlSlashSeparated) {
+      if (isValidLang(part)) {
+        return urlString.replace(`${part}/`, '');
+      }
+    }
```

Comment:
This seems fragile. The locale is the first sub-path of the URL so I'd be in favor of removing only that, instead of looping. AMO URLs cannot be local-less, so I think it'd be fine.

Reply:
I wasn't convinced this was the best route - it was just what I came up with. I can use `url.parse` to extract the `pathName` and then check the first part of that, as you suggest.

Jorge asked that these changes land this week, so they can be in place for the start of the promoted add-ons pilot. This language-stripping part is less important, so I think I'll remove it and file a follow-up so that this can land and we can take more time perfecting this routine.

Reply:
I've deferred these changes to https://github.com/mozilla/addons-frontend/issues/9708